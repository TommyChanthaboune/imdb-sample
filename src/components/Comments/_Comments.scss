
.comments {
  width: 100%;
  h4 {
    font-size: 32px;
  }
}

.comment {
  display: flex;
  border-bottom: 1px solid $greyshie;
  box-shadow: 0px 2px 8px 0px rgba(0, 0, 0, 0.2);
  margin-bottom: 1rem;
  border-radius: functions.toRem(4);
  width: 100%;

  &__profile {
    display: flex;
    flex-flow: column;
    align-items: center;
    width: 100px;
    background-color: $greyshie;
    p {
      font-size: 0.75rem;
      text-align: center;
    }
  }
  &__avatar {
    background-color: $blue;
    width: functions.toRem(50);
    height: functions.toRem(50);
    border-radius: 50%;
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: 1rem;
    img {
      width: functions.toRem(30);
      height: functions.toRem(30);
    }
  }
  &__body {
    padding: 1rem;
    margin: 0 0 1rem;
    padding-bottom: 2rem;
    p {
      margin: 0;
    }
  }

  &__author {
    font-size: 1.25rem;
  }
  &__title {
    font-weight: bold;
  }
}