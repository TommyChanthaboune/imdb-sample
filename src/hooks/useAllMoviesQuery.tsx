import { useQuery } from '@apollo/react-hooks';
import { gql, ApolloError } from '@apollo/client';
import { IMovie } from '../interfaces';

/**
 * NOTE: Due to limitations with JSON-server,
 * this query does not fully reflect each field and
 * associated type, but will return the full set of data
 * regardless.
 * */

const ALL_MOVIES_QUERY = gql`
  query allMovies {
    allMovies {
      id
      title
      voteCount
      voteAverage
      posterPath
      genres
    }
  }
`;

export const useAllMoviesQuery = (): {
  data: IMovie[];
  error: ApolloError | undefined;
  loading: Boolean;
} => {
  const { data, loading, error } = useQuery(ALL_MOVIES_QUERY, {
    fetchPolicy: 'cache-and-network'
  });

  const result = data?.allMovies ?? [];

  return {
    data: result,
    error,
    loading
  };
};
