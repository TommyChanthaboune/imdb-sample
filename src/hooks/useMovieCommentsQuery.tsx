/* eslint-disable @typescript-eslint/no-unused-vars */
import { useEffect } from 'react';
import { useQuery, useMutation } from '@apollo/react-hooks';
import { gql, ApolloError } from '@apollo/client';
import { IComment, IComments, IAddComment, ICommentMeta } from '../interfaces';

interface IMovieCommentsQueryReturn {
  commentData: IComment[];
  commentError: ApolloError | undefined;
  commentLoading: Boolean;
  commentCount: number | undefined;
  commentRefetch: any;
  addComment: any;
}

const COMMENT_QUERY = gql`
  query allComments($movie_id: ID) {
    allComments(filter: { movie_id: $movie_id }, sortOrder: "dsc") {
      id
      title
      author
      data
    }
  }
`;

const COMMENT_META = gql`
  query _allCommentsMeta($movie_id: ID) {
    _allCommentsMeta(filter: { movie_id: $movie_id }) {
      count
    }
  }
`;

const ADD_COMMENT = gql`
  mutation createComment(
    $movie_id: ID!
    $author: String!
    $title: String!
    $data: String!
  ) {
    createComment(
      movie_id: $movie_id
      author: $author
      title: $title
      data: $data
    ) {
      id
    }
  }
`;

export const useMovieCommentsQuery = (
  movie_id: string
): IMovieCommentsQueryReturn => {
  const comment = useQuery<IComments>(COMMENT_QUERY, {
    fetchPolicy: 'cache-and-network',
    variables: {
      movie_id
    }
  });

  const commentMeta = useQuery<ICommentMeta>(COMMENT_META, {
    fetchPolicy: 'cache-and-network',
    variables: {
      movie_id
    }
  });

  const [addComment, { data }] = useMutation<IAddComment, IComment>(
    ADD_COMMENT
  );

  const comments = comment.data?.allComments ?? [];

  return {
    commentData: comments,
    commentError: comment.error,
    commentLoading: comment.loading,
    commentRefetch: comment.refetch,
    // eslint-disable-next-line no-underscore-dangle
    commentCount: commentMeta.data?._allCommentsMeta.count,
    addComment
  };
};
