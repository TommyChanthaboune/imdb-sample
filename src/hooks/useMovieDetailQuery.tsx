/* eslint-disable @typescript-eslint/no-unused-vars */
import { useQuery } from '@apollo/react-hooks';
import { gql, ApolloError } from '@apollo/client';
import { IMovieDetail } from '../interfaces';

interface IMovieDetailQueryReturn {
  data: IMovieDetail;
  error: ApolloError | undefined;
  loading: Boolean;
}

const MOVIE_QUERY = gql`
  query Movie($id: ID!) {
    Movie(id: $id) {
      id
      originalLanguage
      originalTitle
      overview
      popularity
      posterPath
      releaseDate
      title
      voteAverage
      voteCount
      budget
      runtime
      genres
      cast
      director
    }
  }
`;

export const useMovieDetailQuery = (id: string): IMovieDetailQueryReturn => {
  const { data, loading, error } = useQuery(MOVIE_QUERY, {
    fetchPolicy: 'cache-and-network',
    variables: {
      id
    }
  });

  const movie = data?.Movie ?? {};

  return {
    data: movie,
    error,
    loading
  };
};
