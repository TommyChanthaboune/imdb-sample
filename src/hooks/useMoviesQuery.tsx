/* eslint-disable @typescript-eslint/no-unused-vars */
import { useQuery } from '@apollo/react-hooks';
import { gql, ApolloError } from '@apollo/client';
import { IMovieSort, IPaging, IFilterConditons } from '../interfaces';
import { SortOptions } from '../enums';

export interface IMovieQueryReturn {
  data: IMovieSort[];
  error: ApolloError | undefined;
  loading: Boolean;
}

const MOVIE_QUERY = gql`
  query allMovies($sortField: String, $page: Int, $filter: MovieFilter) {
    allMovies(
      sortField: $sortField
      sortOrder: "dsc"
      page: $page
      filter: $filter
    ) {
      id
      title
      voteCount
      voteAverage
      posterPath
      genres
      popularity
      budget
      runtime
      releaseDate
    }
  }
`;

export const useMoviesQuery = (
  paging: IPaging,
  sortOptions: SortOptions,
  filterOptions?: IFilterConditons
): IMovieQueryReturn => {
  const { data, loading, error } = useQuery(MOVIE_QUERY, {
    fetchPolicy: 'cache-and-network',
    variables: {
      sortField: SortOptions[sortOptions],
      filter: filterOptions,
      ...paging
    }
  });

  const result = data?.allMovies ?? [];

  return {
    data: result,
    error,
    loading
  };
};
