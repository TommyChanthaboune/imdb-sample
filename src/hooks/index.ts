import { useAllMoviesQuery } from './useAllMoviesQuery';
import { useMoviesQuery } from './useMoviesQuery';
import { useMovieDetailQuery } from './useMovieDetailQuery';
import { useMovieCommentsQuery } from './useMovieCommentsQuery';

export {
  useAllMoviesQuery,
  useMoviesQuery,
  useMovieDetailQuery,
  useMovieCommentsQuery
};
