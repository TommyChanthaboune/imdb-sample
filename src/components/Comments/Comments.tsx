import React, { FC, useState } from 'react';
import Avatar from '../../assets/Avatar.svg';
import { IComment } from '../../interfaces';
import { useMovieCommentsQuery } from '../../hooks';

interface IComments {
  movie: string;
}

export const Comments: FC<IComments> = ({ movie }) => {
  const [author, setAuthor] = useState<string>('');
  const [title, setTitle] = useState<string>('');
  const [content, setContent] = useState<string>('');
  const [newComments, setNewComments] = useState<IComment[]>([]);

  const {
    commentLoading,
    commentData,
    commentCount,
    addComment
  } = useMovieCommentsQuery(movie);

  const handleAuthorChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setAuthor(e.target.value);
  };
  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setTitle(e.target.value);
  };
  const handleContentChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    setContent(e.target.value);
  };

  const handleAddcomment = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    addComment({
      variables: {
        movie_id: movie,
        author,
        title,
        data: content
      }
    });

    const newComment: IComment = {
      author,
      title,
      data: content,
      id: 'placeholder'
    };

    setNewComments([...newComments, newComment]);
  };

  return (
    <div className="comments">
      <h4>Comments ({(commentCount as number) + newComments.length})</h4>

      <div className="comments__existing">
        {commentLoading ? (
          <h3>Loading Comments</h3>
        ) : (
          <>
            {[...commentData, ...newComments]?.map((comment) => (
              <div className="comment" key={comment.id}>
                <div className="comment__profile">
                  <div className="comment__avatar">
                    <img src={Avatar} alt={comment.author} />
                  </div>
                  <p className="comment__author">{comment.author}</p>
                </div>
                <div className="comment__body">
                  <div className="comment__content">
                    <p className="comment__title">{comment.title}</p>
                    <p className="comment__data">{comment.data}</p>
                  </div>
                </div>
              </div>
            ))}
          </>
        )}
      </div>
      <div className="comments__new">
        <h4>New Comment</h4>
        <form>
          <label htmlFor="comment-author">
            Name
            <input
              type="text"
              id="comment-author"
              onChange={handleAuthorChange}
            />
          </label>
          <label htmlFor="comment-subject">
            Subject
            <input
              type="text"
              id="comment-subject"
              onChange={handleTitleChange}
            />
          </label>
          <label htmlFor="comment-data">
            Comment
            <textarea
              id="comment-data"
              onChange={handleContentChange}
              rows={4}
              cols={40}
            />
          </label>
          <button type="button" onClick={handleAddcomment}>
            Leave Comment
          </button>
        </form>
      </div>
    </div>
  );
};
