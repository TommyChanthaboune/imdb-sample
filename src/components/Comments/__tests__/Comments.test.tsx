import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { Comments } from '../Comments';
import mockData from './mockdata.json';
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { useMovieCommentsQuery } from '../../../hooks/useMovieCommentsQuery';

const mockAddComment = jest.fn();

jest.mock('../../../hooks/useMovieCommentsQuery', () => ({
  useMovieCommentsQuery: () => ({
    commentData: mockData,
    commentError: false,
    commentCount: mockData.length,
    addComment: mockAddComment
  })
}));

describe('Comments', () => {
  it('should render correctly', () => {
    const component = render(<Comments movie="12345" />);

    expect(component).toMatchSnapshot();
  });

  it('should be called with the correct arguements', () => {
    render(<Comments movie="12345" />);
    fireEvent.change(screen.getByLabelText('Name'), {
      target: {
        value: 'test-name'
      }
    });

    fireEvent.change(screen.getByLabelText('Subject'), {
      target: {
        value: 'test-subject'
      }
    });

    fireEvent.change(screen.getByLabelText('Comment'), {
      target: {
        value: 'test-comment'
      }
    });

    fireEvent.click(screen.getByText('Leave Comment'));

    expect(mockAddComment).toBeCalledWith({
      variables: {
        author: 'test-name',
        data: 'test-comment',
        movie_id: '12345',
        title: 'test-subject'
      }
    });
  });
});
