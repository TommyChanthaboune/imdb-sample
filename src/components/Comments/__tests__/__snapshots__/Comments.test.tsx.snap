// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="comments"
      >
        <h4>
          Comments (
          9
          )
        </h4>
        <div
          class="comments__existing"
        >
          <div
            class="comment"
          >
            <div
              class="comment__profile"
            >
              <div
                class="comment__avatar"
              >
                <img
                  alt="author a. komm"
                  src="Avatar.svg"
                />
              </div>
              <p
                class="comment__author"
              >
                author a. komm
              </p>
            </div>
            <div
              class="comment__body"
            >
              <div
                class="comment__content"
              >
                <p
                  class="comment__title"
                >
                  Hey
                </p>
                <p
                  class="comment__data"
                >
                  This is a comment
                </p>
              </div>
            </div>
          </div>
          <div
            class="comment"
          >
            <div
              class="comment__profile"
            >
              <div
                class="comment__avatar"
              >
                <img
                  alt="jr smith"
                  src="Avatar.svg"
                />
              </div>
              <p
                class="comment__author"
              >
                jr smith
              </p>
            </div>
            <div
              class="comment__body"
            >
              <div
                class="comment__content"
              >
                <p
                  class="comment__title"
                >
                  Hey another
                </p>
                <p
                  class="comment__data"
                >
                  This is another comment
                </p>
              </div>
            </div>
          </div>
          <div
            class="comment"
          >
            <div
              class="comment__profile"
            >
              <div
                class="comment__avatar"
              >
                <img
                  alt="author a. komm"
                  src="Avatar.svg"
                />
              </div>
              <p
                class="comment__author"
              >
                author a. komm
              </p>
            </div>
            <div
              class="comment__body"
            >
              <div
                class="comment__content"
              >
                <p
                  class="comment__title"
                >
                  Hey again
                </p>
                <p
                  class="comment__data"
                >
                  This is a third comment
                </p>
              </div>
            </div>
          </div>
          <div
            class="comment"
          >
            <div
              class="comment__profile"
            >
              <div
                class="comment__avatar"
              >
                <img
                  alt="Jon Dough"
                  src="Avatar.svg"
                />
              </div>
              <p
                class="comment__author"
              >
                Jon Dough
              </p>
            </div>
            <div
              class="comment__body"
            >
              <div
                class="comment__content"
              >
                <p
                  class="comment__title"
                >
                  Hello There
                </p>
                <p
                  class="comment__data"
                >
                  This is a short comment
                </p>
              </div>
            </div>
          </div>
          <div
            class="comment"
          >
            <div
              class="comment__profile"
            >
              <div
                class="comment__avatar"
              >
                <img
                  alt="jj"
                  src="Avatar.svg"
                />
              </div>
              <p
                class="comment__author"
              >
                jj
              </p>
            </div>
            <div
              class="comment__body"
            >
              <div
                class="comment__content"
              >
                <p
                  class="comment__title"
                >
                  dean
                </p>
                <p
                  class="comment__data"
                >
                  12345
                </p>
              </div>
            </div>
          </div>
          <div
            class="comment"
          >
            <div
              class="comment__profile"
            >
              <div
                class="comment__avatar"
              >
                <img
                  alt="this is a placeholder"
                  src="Avatar.svg"
                />
              </div>
              <p
                class="comment__author"
              >
                this is a placeholder
              </p>
            </div>
            <div
              class="comment__body"
            >
              <div
                class="comment__content"
              >
                <p
                  class="comment__title"
                >
                  subject
                </p>
                <p
                  class="comment__data"
                >
                  lorem ipsum
                </p>
              </div>
            </div>
          </div>
          <div
            class="comment"
          >
            <div
              class="comment__profile"
            >
              <div
                class="comment__avatar"
              >
                <img
                  alt="another comment"
                  src="Avatar.svg"
                />
              </div>
              <p
                class="comment__author"
              >
                another comment
              </p>
            </div>
            <div
              class="comment__body"
            >
              <div
                class="comment__content"
              >
                <p
                  class="comment__title"
                >
                  comment again
                </p>
                <p
                  class="comment__data"
                >
                  hello
                </p>
              </div>
            </div>
          </div>
          <div
            class="comment"
          >
            <div
              class="comment__profile"
            >
              <div
                class="comment__avatar"
              >
                <img
                  alt="another comment"
                  src="Avatar.svg"
                />
              </div>
              <p
                class="comment__author"
              >
                another comment
              </p>
            </div>
            <div
              class="comment__body"
            >
              <div
                class="comment__content"
              >
                <p
                  class="comment__title"
                >
                  comment again
                </p>
                <p
                  class="comment__data"
                >
                  hello
                </p>
              </div>
            </div>
          </div>
          <div
            class="comment"
          >
            <div
              class="comment__profile"
            >
              <div
                class="comment__avatar"
              >
                <img
                  alt="author a. komm"
                  src="Avatar.svg"
                />
              </div>
              <p
                class="comment__author"
              >
                author a. komm
              </p>
            </div>
            <div
              class="comment__body"
            >
              <div
                class="comment__content"
              >
                <p
                  class="comment__title"
                >
                  Hey again
                </p>
                <p
                  class="comment__data"
                >
                  This is a third comment
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="comments__new"
        >
          <h4>
            New Comment
          </h4>
          <form>
            <label
              for="comment-author"
            >
              Name
              <input
                id="comment-author"
                type="text"
              />
            </label>
            <label
              for="comment-subject"
            >
              Subject
              <input
                id="comment-subject"
                type="text"
              />
            </label>
            <label
              for="comment-data"
            >
              Comment
              <textarea
                cols="40"
                id="comment-data"
                rows="4"
              />
            </label>
            <button
              type="button"
            >
              Leave Comment
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="comments"
    >
      <h4>
        Comments (
        9
        )
      </h4>
      <div
        class="comments__existing"
      >
        <div
          class="comment"
        >
          <div
            class="comment__profile"
          >
            <div
              class="comment__avatar"
            >
              <img
                alt="author a. komm"
                src="Avatar.svg"
              />
            </div>
            <p
              class="comment__author"
            >
              author a. komm
            </p>
          </div>
          <div
            class="comment__body"
          >
            <div
              class="comment__content"
            >
              <p
                class="comment__title"
              >
                Hey
              </p>
              <p
                class="comment__data"
              >
                This is a comment
              </p>
            </div>
          </div>
        </div>
        <div
          class="comment"
        >
          <div
            class="comment__profile"
          >
            <div
              class="comment__avatar"
            >
              <img
                alt="jr smith"
                src="Avatar.svg"
              />
            </div>
            <p
              class="comment__author"
            >
              jr smith
            </p>
          </div>
          <div
            class="comment__body"
          >
            <div
              class="comment__content"
            >
              <p
                class="comment__title"
              >
                Hey another
              </p>
              <p
                class="comment__data"
              >
                This is another comment
              </p>
            </div>
          </div>
        </div>
        <div
          class="comment"
        >
          <div
            class="comment__profile"
          >
            <div
              class="comment__avatar"
            >
              <img
                alt="author a. komm"
                src="Avatar.svg"
              />
            </div>
            <p
              class="comment__author"
            >
              author a. komm
            </p>
          </div>
          <div
            class="comment__body"
          >
            <div
              class="comment__content"
            >
              <p
                class="comment__title"
              >
                Hey again
              </p>
              <p
                class="comment__data"
              >
                This is a third comment
              </p>
            </div>
          </div>
        </div>
        <div
          class="comment"
        >
          <div
            class="comment__profile"
          >
            <div
              class="comment__avatar"
            >
              <img
                alt="Jon Dough"
                src="Avatar.svg"
              />
            </div>
            <p
              class="comment__author"
            >
              Jon Dough
            </p>
          </div>
          <div
            class="comment__body"
          >
            <div
              class="comment__content"
            >
              <p
                class="comment__title"
              >
                Hello There
              </p>
              <p
                class="comment__data"
              >
                This is a short comment
              </p>
            </div>
          </div>
        </div>
        <div
          class="comment"
        >
          <div
            class="comment__profile"
          >
            <div
              class="comment__avatar"
            >
              <img
                alt="jj"
                src="Avatar.svg"
              />
            </div>
            <p
              class="comment__author"
            >
              jj
            </p>
          </div>
          <div
            class="comment__body"
          >
            <div
              class="comment__content"
            >
              <p
                class="comment__title"
              >
                dean
              </p>
              <p
                class="comment__data"
              >
                12345
              </p>
            </div>
          </div>
        </div>
        <div
          class="comment"
        >
          <div
            class="comment__profile"
          >
            <div
              class="comment__avatar"
            >
              <img
                alt="this is a placeholder"
                src="Avatar.svg"
              />
            </div>
            <p
              class="comment__author"
            >
              this is a placeholder
            </p>
          </div>
          <div
            class="comment__body"
          >
            <div
              class="comment__content"
            >
              <p
                class="comment__title"
              >
                subject
              </p>
              <p
                class="comment__data"
              >
                lorem ipsum
              </p>
            </div>
          </div>
        </div>
        <div
          class="comment"
        >
          <div
            class="comment__profile"
          >
            <div
              class="comment__avatar"
            >
              <img
                alt="another comment"
                src="Avatar.svg"
              />
            </div>
            <p
              class="comment__author"
            >
              another comment
            </p>
          </div>
          <div
            class="comment__body"
          >
            <div
              class="comment__content"
            >
              <p
                class="comment__title"
              >
                comment again
              </p>
              <p
                class="comment__data"
              >
                hello
              </p>
            </div>
          </div>
        </div>
        <div
          class="comment"
        >
          <div
            class="comment__profile"
          >
            <div
              class="comment__avatar"
            >
              <img
                alt="another comment"
                src="Avatar.svg"
              />
            </div>
            <p
              class="comment__author"
            >
              another comment
            </p>
          </div>
          <div
            class="comment__body"
          >
            <div
              class="comment__content"
            >
              <p
                class="comment__title"
              >
                comment again
              </p>
              <p
                class="comment__data"
              >
                hello
              </p>
            </div>
          </div>
        </div>
        <div
          class="comment"
        >
          <div
            class="comment__profile"
          >
            <div
              class="comment__avatar"
            >
              <img
                alt="author a. komm"
                src="Avatar.svg"
              />
            </div>
            <p
              class="comment__author"
            >
              author a. komm
            </p>
          </div>
          <div
            class="comment__body"
          >
            <div
              class="comment__content"
            >
              <p
                class="comment__title"
              >
                Hey again
              </p>
              <p
                class="comment__data"
              >
                This is a third comment
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="comments__new"
      >
        <h4>
          New Comment
        </h4>
        <form>
          <label
            for="comment-author"
          >
            Name
            <input
              id="comment-author"
              type="text"
            />
          </label>
          <label
            for="comment-subject"
          >
            Subject
            <input
              id="comment-subject"
              type="text"
            />
          </label>
          <label
            for="comment-data"
          >
            Comment
            <textarea
              cols="40"
              id="comment-data"
              rows="4"
            />
          </label>
          <button
            type="button"
          >
            Leave Comment
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
